src/LockstakeEngine.sol

```diff
@@ -172,8 +169,18 @@ contract LockstakeEngine is Multicall {
         }
     }
 
-    function _urnAuth(address urn, address usr) internal view returns (bool ok) {
-        ok = urnOwners[urn] == usr || urnCan[urn][usr] == 1;
+    function _urnAuth(address owner, address urn, address usr) internal view returns (bool ok) {
+        ok = owner == usr || urnCan[urn][usr] == 1;
+    }
+
+    function _getUrn(address owner, uint256 index) internal view returns (address urn) {
+        urn = ownerUrns[owner][index];
+        require(urn != address(0), "LockstakeEngine/invalid-urn");
```

Comment:
just curious, if you remove the check for `urn != address(0)` will something break?
It is probably good to have it anw, I just want to make sure i don't miss anything in the threat model.

Reply:
As mentioned in the prev comment, not really breaking, but it is a nice check when `lock`ing. And in general it kinda feels good to first check something is valid before checking its permissions, but yeah far from being mandatory or similar.